### DISCUSSION TITLE
How to maintain whitespaces in for-loops

### DISCUSSION THREAD

**Discussion by HenrikPoulsen** [harttle/liquidjs#354] [Q&A]
Hello.
I'm working on something where I need to parse yaml files that use liquid. I have done some experimentation with liquidjs but I pretty quickly got into a situation where it seems to clobber leading whitespaces, which ends up making it invalid yaml.
Here's a simplified example:
```js
await engine.parseAndRender(`
dependencies:
    - a hardcoded thing
{% for thing in things -%}
    - {{ thing }}
{% endfor -%}
    `, {things: [1, 2, 3, 4]})
```
The expected output here is:
```yml
 dependencies:
     - a hardcoded thing
     - 1
     - 2
     - 3
     - 4
```
but it becomes
```yml
dependencies:
    - a hardcoded thing
- 1
- 2
- 3
- 4

```

[Here's](https://liquidjs.com/playground.html#eyUgYXNzaWduIHRoaW5ncyA9ICJhbGljZSwgYm9iLCBjYXJvbCIgfCBzcGxpdDogIiwgIiAtJX0KCmRlcGVuZGVuY2llczoKICAgIC0gYSBoYXJkY29kZWQgdGhpbmcKeyUgZm9yIHRoaW5nIGluIHRoaW5ncyAtJX0KICAgIC0ge3sgdGhpbmcgfX0KeyUgZW5kZm9yIC0lfQo=) an example using the playground which shows the same.
How can I avoid this behavior without requiring people to change their yml?
Is this a bug or can I configure this somehow?

**Comment by harttle** [comment]
The behavior of `-` is hard to understand even for me. There're `greedy` options to twick it a bit. Try that!

**Reply by pdehaan** [reply]
How about `{%-` instead of `-%}`?

```js
{% assign things = "alice, bob, carol" | split: ", " -%}

dependencies:
    - a hardcoded thing
{%- for thing in things %}
    - {{ thing }}
{%- endfor %}
```

### OUTPUT (from playground)

```md
dependencies:
    - a hardcoded thing
    - alice
    - bob
    - carol
```

Only other idea I had was don't use `{%-` _or_ `-%}` and just have some regexp or something that removes empty lines of whitespace on the rendered YAML output, but that feels risky.
